packages/next-app/src/app/stableswapsimulator/(components)/VariantDataForm.tsx

```diff
@@ -0,0 +1,86 @@
+"use client";
+
+import { StableSwapSimulatorDataSchema } from "@balancer-pool-metadata/schema";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useEffect } from "react";
+import { useForm } from "react-hook-form";
+
+import { Input } from "#/components/Input";
+import { Select, SelectItem } from "#/components/Select";
+import { AnalysisData, useStableSwap } from "#/contexts/StableSwapContext";
+
+export default function NewDataForm() {
+  const {
+    baselineData,
+    variantData,
+    setVariantData,
+    setIndexAnalysisToken,
+    indexAnalysisToken,
+  } = useStableSwap();
+
+  const {
+    register,
+    setValue,
+    getValues,
+    formState: { errors },
+  } = useForm<typeof StableSwapSimulatorDataSchema._type>({
+    resolver: zodResolver(StableSwapSimulatorDataSchema),
+    mode: "onChange",
+  });
+
+  const handleChange = () => {
+    if (Object.keys(errors).length) return;
+    const data = getValues();
+    setVariantData(data as AnalysisData);
+  };
```

Comment:
Shouldn't we just handle onSubmit as the user hits enter/clicks to refresh the simulation here?

Reply:
Think that we didn't discuss this that much but my idea was the simulation refresh on change of each input. What do you think about it?

Reply:
If that's not computationally intensive I think it would work -- just wonder if it won't be to jittery -- but I think you added the debounced values because of this. Personally I'd not reimplement the form logic and just call submit (debounced, perhaps) after an input changes and I wanted the simulations to be refreshed.